tests/test_core.py

```diff
@@ -6050,7 +6054,10 @@ def test_poppler(self):
   @is_slow_test
   def test_openjpeg(self):
     if '-fsanitize=address' in self.emcc_args:
-      self.set_setting('INITIAL_MEMORY', 128 * 1024 * 1024)
+      # In ASan mode we need a large initial memory (or else wasm-ld fails).
+      # The OpenJPEG CMake will build several executables, so we must enable
+      # the flag for them all.
+      os.environ['EMMAKEN_CFLAGS'] = '-sINITIAL_MEMORY=300MB'
```

Comment:
Ooo.. can't we use `self.set_setting` here?

Reply:
No, because those are not sent to all the things cmake drives. Our `set_setting` stuff only applies for the `emcc` invocations we do directly.

Reply:
I'm really not a fan of these backdoor method of injecting flags.

I'm a little confused though.. isn't openjpeg a library rather than an exectuable?  Don't we build the executable directly below?  (and isn't `-sINITIAL_MEMORY` irrelevant unless you are linking an executable).

Reply:
Yeah, I don't like it either.

It is a library, but it also builds some commandline tools that use the library. We actually use those in our testing. So we could in theory refactor the test to build those manually ourselves, perhaps.

Reply:
If you do do it this way can you limit the scope with `with modify_env:` block?

Reply:
If `set_setting` doesn't work that looks like a bug because `get_library` pass `cflags=self.get_emcc_args()` to `build_library`.

Perhaps the problem is that this is a link flag not a cflag?

Reply:
Oh I saw this after I landed. I'll add an env block.

And yes, good point, maybe the issue is that we need this at link time.